pkg/reconciler/common/transformers.go

```diff
@@ -545,32 +541,42 @@ func AddConfigMapValues(configMapName string, prop interface{}) mf.Transformer {
 		}
 
 		values := reflect.ValueOf(prop)
-		types := values.Type()
+		// if the given properties is not struct type, do not proceed
+		if values.Kind() != reflect.Struct {
+			return nil
+		}
 
-		for i := 0; i < values.NumField(); i++ {
-			key := strings.Split(types.Field(i).Tag.Get("json"), ",")[0]
+		for index := 0; index < values.NumField(); index++ {
+			key := strings.Split(values.Type().Field(index).Tag.Get("json"), ",")[0]
 			if key == "" {
 				continue
 			}
-			if values.Field(i).Kind() == reflect.Ptr {
-				innerElem := values.Field(i).Elem()
 
-				if !innerElem.IsValid() {
-					continue
-				}
+			element := values.Field(index)
+			if element.Kind() == reflect.Ptr {
+				// extract the actual element from the pointer
+				element = values.Field(index).Elem()
+			}
 
-				if innerElem.Kind() == reflect.Bool {
-					cm.Data[key] = strconv.FormatBool(innerElem.Bool())
-				} else if innerElem.Kind() == reflect.Uint {
-					cm.Data[key] = strconv.FormatUint(innerElem.Uint(), 10)
-				} else if innerElem.Kind() == reflect.String {
-					cm.Data[key] = innerElem.String()
-				}
+			if !element.IsValid() {
 				continue
 			}
 
-			if value := values.Field(i).String(); value != "" {
```

Comment:
is this case handled ?

Reply:
@piyush-garg Sorry I have changed many thing on this PR. It should be WIP for now.
I am sorry for that. I will inform your when it is ready to review.

Reply:
@piyush-garg Now it is ready to review. can you please review the PR?